students/javarb/sessions/2019/april/Session_30.md

```diff
@@ -0,0 +1,108 @@
+# SOFTWARE ENGINEERING IN JAVA
+
+## Session 30 (16/04/2019)
+
+- Reviewing sprint project (issue) tasks and solving problems and feedback
+- Reviewing main project tasks and planning next Sprint
+
+## Notes
+
+#### Reviewing sprint project (issue) tasks and solving problems and feedback
+
+#### Unit Tests
+
+Test for service were created and controller test which I had done, were improved.
+
+In controller test, we removed these annotations because they are used when we're creating Integration Tests when we need to run whole application in Controller. :
+
+```java
+@RunWith(SpringRunner.class)
+@SpringBootTest
+```
+
+Integration Tests run the entire application and because that, take more time to run while Unit Test are just loading Junit which usually takes less than 1 sec. Unit test are used to simulate classes of which depends what is being tested. 
```

Comment:
```suggestion
Integration Tests run the entire application and because of that, take more time to run while Unit Test are just loading Junit which usually takes less than 1 sec. Unit test are used to simulate classes of which depends what is being tested. 
```